src/pages/smart-contracts/stacks-api-client-tutorial.md

```diff
@@ -0,0 +1,727 @@
+          Recent Activities:{' '}
+          {activities.map((activity, key) => {
+            <Text key={key}>{JSON.stringify(activity)}</Text>;
+          })}
+        </Text>
+      </Box>
+    </Flex>
+  ) : null;
+};
+```
+
+Then include it in the app by add the following lines to `Apps.jsx` after the `SignIn` and `TodoList` fragment:
+
+```js
+<RecentActivities />
+```
+
+### 2. Step: Use AccountsApi
+
+Now, add a call to the `getAccountTransactions` method. This method can be used for users and for contracts. It returns transactions that are related to the given account. The default is to return the last 20 transactions.
+
+```js
+const fetchActivities = useCallback(async () => {
+  // fetch activities
+  const response = await accountsApi.getAccountTransactions({
+    principal: `${CONTRACT_ADDRESS}.${CONTRACT_NAME}`,
+  });
+  console.log(response);
+}, []);
+```
+
+-> For users, the principal argument is just user's the stacks address.
+
+### 3. Step: Filter for successful register transactions
+
+The returned transactions can be successful or failed `register` or `update` calls, or even the contract deployment transaction. The recent activities should only contain successfull contract calls. Therefore, add a filter function
+
+```js
+const contractCallsOnly = r => {
+  console.log(r);
+  return r.tx_status === 'success' && r.tx_type === 'contract_call';
+};
+```
+
+and apply it to the response result:
+
+```js
+setRegistrations(response.results.filter(contractCallsOnly));
+```
+
+The final callback `fetchActivities` looks like this:
+
+```js
+const fetchActivities = useCallback(async () => {
+  // fetch activities
+  const response = await accountsApi.getAccountTransactions({
+    principal: `${CONTRACT_ADDRESS}.${CONTRACT_NAME}`,
+  });
+  console.log(response);
+  setRegistrations(response.results.filter(contractCallsOnly));
+}, []);
+```
+
+You should now see a list of hex strings representing the transactions.
+
+### 4. Step: Extract registry IDs
+
+The last step is to create a nice representation of the transactions. The `register` function returns the transaction id as a result. As mentioned above, it is an unsigned int (`uint`). The api call returns the transaction result as a hex string (`tx_result.hex`) like `0x0100000000000000000000000000000001`. It represents the serialized clarity value in hex encoding. The method `deserializeCV` creates a ClarityValue object out of the hex string, only the leading `0x` needs to be removed. The following line returns a `UintCV` object for the `register` transaction:
+
+```js
+deserializeCV(Buffer.from(activity.tx_result.hex.substr(2), 'hex'));
+```
+
+The client library defines types for all Clarity types. The corresponding type gives access to more properties. For example, `UintCV` has a `value` property for its number value:
+
+```js
+cv.value;
+```
+
+or you can use `cvToString` for an opinionated conversion to a string
```

Comment:
What should the reader think of if we say "opinionated conversion"? Would give an example and be clear to avoid confusion

Reply:
using "simple conversion"